app/code/Magento/CatalogGraphQl/Service/CheckImageCacheFileService.php

```diff
@@ -0,0 +1,123 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\CatalogGraphQl\Service;
+
+use Magento\Catalog\Model\Product\Image;
+use Magento\Catalog\Model\Product\ImageFactory;
+use Magento\Catalog\Model\Product\Image\ParamsBuilder;
+use Magento\Framework\App\Filesystem\DirectoryList;
+use Magento\Framework\Exception\FileSystemException;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Framework\Filesystem;
+use Magento\Store\Model\StoreManagerInterface;
+
+/**
+ * Service check if image cache file exist
+ */
+class CheckImageCacheFileService
```

Comment:
I would suggest clarifying the name of the service since "Check" might be ambiguous. Considering the fact that Magento core does not contain services in its classic DDD definition, I would suggest moving the class to the Model directory and renaming it in the following way 

```suggestion
class CheckImageCacheFileExist
```